File: 카카오 겨울인턴2019/징검다리 건너기.py
def is_poss(middle, stones, k):
    empty = 0
    for stone in stones:
        if stone - middle < 0:
            empty += 1
            if empty == k:
                return False
        else:
            empty = 0
    return True

def solution(stones, k):
    left, right = min(stones), max(stones)
    
    while left < right: # 최댓값 탐색
        middle = (left + right + 1) // 2 # 주의, 최솟값 탐색과 비교
        if is_poss(middle, stones, k):
            left = middle
        else:
            right = right - 1
            
    return right

File: 카카오 겨울인턴2019/징검다리 건너기.js
function solution(stones, k) {
    let left = 1 // Math.min(...stones)
    let right = 200000000 // Math.max(...stones)
    while (left < right) {
        let middle = parseInt((left + right + 1) / 2) // 나눗셈
        if (is_poss(middle, stones, k)) {
            left = middle
        } else right = middle - 1
    }
    return right
}

function is_poss(middle, stones, k) {
    let empty = 0
    for (const stone of stones) {
        if (stone - middle < 0) {
            empty += 1
            if (empty === k) return false
        } else empty = 0
    }
    return true
}